### DISCUSSION TITLE
Audio Tools on Arduino 1.8.19

### DISCUSSION THREAD

**Discussion by geoffw64** [pschatzmann/arduino-audio-tools#1562] [Q&A] [answered]
Hi Phil, 
Hope you're well.    
I downloaded arduino-audio-tools-0.9.8.zip from github and unzipped it into my Arduino libraries folder.  My environment:
Mac OS/X 10.13.  Using this old version as my old Mac can't be upgraded further. 
Adafruit ESP32 board with WROOM chip. 
Arduino 1.8.19.  I understand this can't be upgraded to 2.0 on this version of OS/X.
ESP_IDF 3.3  

This sketch... 
#include "Arduino.h"
#include "AudioTools.h"
void setup() {
}
void loop() {
}
produces the following compile error...
In file included from /Users/geoffw/Documents/Arduino/libraries/arduino-audio-tools/src/Concurrency/Concurrency.h:9:0,
                 from /Users/geoffw/Documents/Arduino/libraries/arduino-audio-tools/src/AudioTools.h:67,
                 from /Users/geoffw/Documents/Arduino/AudioToolsTest/AudioToolsTest.ino:2:
/Users/geoffw/Documents/Arduino/libraries/arduino-audio-tools/src/Concurrency/BufferRTOS.h:9:38: fatal error: freertos/stream_buffer.h: No such file or directory
compilation terminated.
exit status 1
Error compiling for board Adafruit ESP32 Feather.

It can't find freertos/stream_buffer.h apparently because there is no freertos folder in the arduino-audio-tools hierarchy.  Is this meant to be included in the src folder of arduino-audio-tools or do I need to install something else? 
Hierarchy of the arduino-audio-tools folder on my Mac is attached.     
![Screen Shot 2024-05-15 at 12 21 40 pm](https://github.com/pschatzmann/arduino-audio-tools/assets/163606841/d5d87785-906f-49b3-ae5f-5ec405d48cb1)

**Comment by pschatzmann** [comment, marked as answer]
Please double check the ESP32 core version: The actual version is 2.0.16!
FreeRTOS is part of the IDF which is part of the Arduino ESP32 core.

**Reply by geoffw64** [reply]
Thanks Phil.
Tried to install version 2.x of the esp32 core but it seems to require python 3 which in turn requires a later version of OS/X than I can install on the old Mac. 
Not to worry, I'll try using your arduino-libFLAC library directly.  I'm decoding FLAC to PCM over i2s. 
I'll find a reason to replace my old Mac at some point.  
Thanks again for responding. 
Cheers, Geoff.

**Reply by pschatzmann** [reply]
I was in the same situation and got tired to patch the installations to make it work, so I installed Linux on my old Mcbooks.

I also committed a correction to make old versions compile again: I tested it with the old 1.0.6 ESP core version

**Reply by geoffw64** [reply]
Good use for old Macs.   They just don’t die.   I imagine they would run Linux really well.  
Thank you so much for updating the library to work on earlier Arduino core versions.  I’ll download and give it a test.

**Reply by geoffw64** [reply]
Thanks Phil; now compiles cleanly on my setup.  
Cheers, Geoff
